### ISSUE TITLE
fix(aws): change the `RESTMethodSettings` type to a list of settings

### ISSUE THREAD

**Issue by nikpivkin** [aquasecurity/trivy#6432]
1. Several settings can be [specified](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-api.html#sam-api-methodsettings) for the API stage.
2. This is only supported for CloudFormation, but Terraform also [has](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_method_settings#settings) a resource for managing API settings.

**Comment by itaysk** [comment]
is this about changing the input structure of the check? if so 1) need to update the schema as well 2) isn't it a breaking change?
cc @simar7

**Comment by nikpivkin** [comment]
@itaysk 
Yes, this is an update to the input data for the checks.
1. Yes
2. Most likely if users use this structure in custom checks. Should the title be updated?

**Comment by itaysk** [comment]
yes, the title should be updated, and we should remember to mention it in the release notes.

also, I think we should be more minded that changing input structure is a breaking change and follow a procedure:
1. explain why the change is needed
2. is it possible to avoid it
4. what is the impact / recovery method
3. give some notice in advance, and collect feedback
4. gradually roll out

I understand this might be a low impact change so we can fast-forward through some of the process, but we should at least be intentional about it

**Comment by simar7** [comment]
@itaysk @nikpivkin we'll err on the side of caution and mark this as a breaking change. We don't have usage metrics so can't articulate how many users it will affect or how widely it is used (it's a non required argument).
